build-tools/build.gradle.kts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.util.Properties
+
+plugins {
+    // We explicitly configure versions of plugins in settings.gradle.kts.
+    // due to https://github.com/gradle/gradle/issues/1697.
+    id("kotlin")
+    id("kotlinx-serialization")
+    groovy
+    `java-gradle-plugin`
+}
+
+val rootProperties = Properties().apply {
+    rootDir.resolve("../gradle.properties").reader().use(::load)
```

Comment:
Is this correct? Isn't the `rootDir` points to the whole project's root?
If so then `"../gradle.properties"` seems to be incorrect

Reply:
No, `rootDir` points to the `build-tools` directory because `build-tools` is a separate build included into the main one thus it have its own root directory.